GIScience/openrouteservice
# Fix docker: dynamic app.config and dockerhub integration

### Pull Request Checklist

- [x] 1. I have [**rebased**](https://github.com/GIScience/openrouteservice/blob/master/CONTRIBUTE.md#pull-request-guidelines) the latest version of the master branch into my feature branch and all conflicts have been resolved.
- [x] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [ ] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ ] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the app.config file, I have added these to the app.config wiki page on github along with a short description of what it is for, and documented this in the Pull Request (below).
- [ ] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [x] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [ ] 10. For new features or changes involving building of graphs, I have tested on a larger dataset (at least Germany) and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the importer etc.), I have generated longer distance routes for the affected profiles with different options (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end points generated from the current live ORS. If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [x] 12. I have written in the Pull Request information about the changes made including their intended usage and why the change was needed.

Fixes #696 , EDIT: fixes #718 as well.

### Information about the changes
- Key functionality added: 
  - changes to host's `app.config` are reflected now inside the container
  - `CATALINA/JAVA_OPTS` are set correctly
  - `app.config` doesn't have to be present when starting a container, which finally makes it independent of this repository and fully available for Dockerhub images

### Things to test

1. Custom `app.config`:
  - build image locally and run a container in a directory other than this repository (e.g. `/tmp`)
  - change `app.config.sample` which is now supposed to be mapped to `./conf/app.config.sample`
  - restart container and confirm changes took effect

2. use other PBF as a volume and `BUILD_GRAPHS=True`, which should delete `./graphs/*` and regenerate
